lib/order.rb

```diff
@@ -1,18 +1,90 @@
+require 'csv'
+require 'awesome_print'
+
+
 module Grocery
   class Order
     attr_reader :id, :products
 
+
     def initialize(id, products)
       @id = id
       @products = products
     end
 
-    def total
-      # TODO: implement total
+
+    def self.all
+      orders = []
+      # "This is a class method."
+      # This is a class method to access the class variabel
+      #returns the csv file to an array with headers
+      #orders_csv = CSV.read('support/orders.csv', 'r')
+      orders_csv = CSV.read("support/orders.csv", 'r',headers: true).to_a
+
+
+      orders_csv.each do |line|
+        orders << Grocery::Order.new(*line)
+      end
+      return orders
+    end
+
+
+    def self.find(id)
+      a = self.all
+
+      if id > a.length
+        return nil
+      end
```

Comment:
This implementation assumes that the IDs in the file are sequential. On this project that ends up being true, but as we get into Rails and look at more complex database setups, this will not always be the case.

A more flexible implementation of this method might look like:

```ruby
def self.find(id)
  all.each do |order|
    return order if order.id == id
  end
  return nil
end
```

Reply:
Thanks! I will keep that in mind for the future